### PR TITLE
developing of the new database schema for the faq_ tables; editing of the existing API to match the modifications

### PR DESCRIPTION
**Pull request by raynor85**
developing of the new database schema for the faq_ tables; editing of the existing API to match the modifications

### CODE REVIEW

**Comment by zbirmingham** [comment]
@young @obraafo can we get a code check on this?

File: index.php [hunk 1]
```diff
@@ -271,7 +271,7 @@ function getAppConfigFile($configFile) {
 	}
 
 	if (isset($occupationIds) && count($occupationIds) > 0) {
-		$questions = executeSql(
+		/*$questions = executeSql(
```
Comment (young) [inline index.php]: Delete the commented out block

File: index.php [hunk 2]
```diff
@@ -285,9 +285,22 @@ function getAppConfigFile($configFile) {
...
+					INNER JOIN faq_response ON
+						faq_response.FAQ_QuestionId = fq.Id
+				WHERE fq.OccupationId IN (:occupationIds)
+				ORDER BY fq.Text ASC
+			',
+			['occupationIds' => implode(',', $occupationIds)]
 		);
 	} else {
-		$questions = executeSql('
+		/*$questions = executeSql('
```
Comment (young) [inline index.php]: Delete the commented out block

File: index.php [hunk 3]
```diff
@@ -343,7 +365,7 @@ function getAppConfigFile($configFile) {
 
 $app->get('/questions/search/:searchQuery', function($searchQuery) use($app) {
     // find matching questions
-    $questions = executeSql(
+    /*$questions = executeSql(
```
Comment (young) [inline index.php]: Same as above

File: index.php [hunk 4]
```diff
@@ -369,7 +404,7 @@ function getAppConfigFile($configFile) {
     }
 
     // find matching answers
-    $answers = executeSql(
+    /*$answers = executeSql(
```
Comment (young) [inline index.php]: Same as above

File: index.php [hunk 5]
```diff
@@ -407,7 +457,7 @@ function getAppConfigFile($configFile) {
 });
 
 $app->get('/questions/:id/answers', function($id) use ($app) {
-	$answers = executeSql(
+	/*$answers = executeSql(
```
Comment (young) [inline index.php]: Same

File: expert_operations.php [hunk 6]
```diff
@@ -0,0 +1,162 @@
+<?php
+
```
Comment (young) [inline expert_operations.php:2]: Since this is a different file belonging to the API, can you add a quick summary of what this file is for?

File: expert_operations.php [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+/**
+ * A function that will accept a like to to a specific response. (Ruben)
+ *
+ * @param int $e_id id of the expert, e.g. 5023.
+ * @param int $r_id id of the response of another expert, e.g. 633491.
+ *
+ * @return none
+ */
+
+$app->get('/add_like_response/:e_id/:r_id', function($e_id, $r_id) use ($app) {
```
Comment (young) [inline expert_operations.php:155]: Are we going to have a function that gets the most liked questions?

**Comment by young** [comment]
Looks pretty good. One thing I'd like to address now or in the near future is the naming of the database tables. We have table like 'faq_questionassignment' and it'd be better to either underscore the entire name or camel case it for readability.

File: expert_operations.php (hunk 7 above)
Comment (raynor85) [inline expert_operations.php:155]: That API just add like to a response not to a question. I have no idea
about likes related to questions.

2014-11-10 17:34 GMT-05:00 Jemuel Young notifications@github.com:

**Comment by raynor85** [comment]
I totally agree, but I kept the existing database as similar as possible to
minimize preexisting API corruption.

2014-11-10 17:38 GMT-05:00 Jemuel Young notifications@github.com:

File: expert_operations.php (hunk 7 above)
Comment (young) [inline expert_operations.php:155]: Sorry, I meant responses, not questions. This function adds the ability to 'like' something so there _should_ be some way of pulling those likes back to the frontend.

**Comment by young** [comment]
It might be worth addressing now or in the next PR so it doesn't continue to be in the next PR.

**Comment by young** [comment]
Other than the comments I added, looks good. Ship it.

File: expert_operations.php (hunk 7 above)
Comment (raynor85) [inline expert_operations.php:155]: The # of like related to a response is represented by the number of
rows having same FAQ_ResponseId foreign key. A simple select query can
retrieve the number of like of that response, and the unicity of the
primary key allows implicitly to allow only one like for every expert.
It also allows the expert to remove its like choice.
To retrieve the data we can create an inner query that count the
number of responses and pass the output to the parent query that
basically add the result for every response rows.

2014-11-10 17:42 GMT-05:00 Jemuel Young notifications@github.com:

File: expert_operations.php (hunk 6 above)
Comment (raynor85) [inline expert_operations.php:2]: I had deleted them. The file has a different name, but conceptually is the
same as the previous one. I changed the name because it matches better than
the previous one. This file includes the operations of the expert in their
dashboard.

2014-11-10 17:23 GMT-05:00 Jemuel Young notifications@github.com:

Comment (young) [inline expert_operations.php:2]: Can you add that into the file? I want to improve some of our documentation.

Comment (raynor85) [inline expert_operations.php:2]: What is it the meaning of that?
Also index.php is without...

2014-11-10 18:08 GMT-05:00 Jemuel Young notifications@github.com:

Comment (young) [inline expert_operations.php:2]: Yeah index.php is probably too large to add a quick summary but this file is pretty concise and specific that it could use summary of the form:

```
/**
 * Short Description
 *
 * Long Description
 *
 */
```

[Link](http://pear.php.net/manual/en/standards.sample.php)
